src/js/internal/sql/sqlite.ts

```diff
@@ -0,0 +1,611 @@
+import type * as BunSQLiteModule from "bun:sqlite";
+import type { BaseQueryHandle, Query, SQLQueryResultMode } from "./query";
+import type { DatabaseAdapter, OnConnected, SQLHelper, SQLResultArray } from "./shared";
+
+const { SQLHelper, SQLResultArray } = require("internal/sql/shared");
+const {
+  Query,
+  SQLQueryResultMode,
+  SQLQueryFlags,
+  symbols: { _strings, _values },
+} = require("internal/sql/query");
+const { escapeIdentifier, connectionClosedError } = require("internal/sql/utils");
+const { SQLiteError } = require("internal/sql/errors");
+
+let lazySQLiteModule: typeof BunSQLiteModule;
+function getSQLiteModule() {
+  if (!lazySQLiteModule) {
+    lazySQLiteModule = require("../../bun/sqlite.ts");
+  }
+  return lazySQLiteModule;
+}
+
+enum SQLCommand {
+  insert = 0,
+  update = 1,
+  updateSet = 2,
+  where = 3,
+  whereIn = 4,
+  none = -1,
+}
+
+function commandToString(command: SQLCommand): string {
+  switch (command) {
+    case SQLCommand.insert:
+      return "INSERT";
+    case SQLCommand.updateSet:
+    case SQLCommand.update:
+      return "UPDATE";
+    case SQLCommand.whereIn:
+    case SQLCommand.where:
+      return "WHERE";
+    default:
+      return "";
+  }
+}
+
+function detectCommand(query: string): SQLCommand {
```

Comment:
This causes it to duplicate the query string potentially multiple times.

Instead, we can loop until we have a space-like delimiter and then .slice() just that region?